src/Entity/ItemId.php

```diff
@@ -103,4 +103,21 @@ public static function newFromNumber( $numericId ) {
 		return new self( 'Q' . $numericId );
 	}
 
+	/**
+	 * CAUTION: Use the full string serialization whenever you can and avoid using numeric IDs.
+	 *
+	 * @param string $repositoryName
+	 * @param int|float|string $numericId
+	 *
+	 * @return self
+	 * @throws InvalidArgumentException
+	 */
+	public static function newForeignFromNumber( $repositoryName, $numericId ) {
```

Comment:
I feel the name "**foreign** from number" hints that the "foreign" bit comes first. This is also consistent with the static `EntityId::…Serialization` methods.